includes/utilities.php

```diff
@@ -85,3 +85,57 @@ function prefix_path( $url ) {
 
 	return $url;
 }
+
+/**
+ * Add a redirect.
+ *
+ * @param array $redirect Arguments for the redirect.
+ *
+ * @return int The ID of the Post if redirect successfully added. Otherwise returns 0.
+ */
+function insert_redirect( $redirect ) {
+	// Stop loops.
+	remove_action( 'save_post', 'HM\Redirects\\Admin_UI\\handle_redirect_saving', 13 );
+
+	if ( ! isset( $redirect['post_id'] ) ) {
+		$redirect['post_id'] = 0;
+	}
+
+	$post_id = wp_insert_post( [
+		'ID'                    => $redirect['post_id'],
+		'post_content_filtered' => strtolower( $redirect['status_code'] ),
+		'post_excerpt'          => strtolower( $redirect['to'] ),
+		'post_name'             => get_url_hash( $redirect['from'] ),
+		'post_status'           => 'publish',
+		'post_title'            => strtolower( $redirect['from'] ),
+		'post_type'             => REDIRECTS_POST_TYPE,
+	] );
+
+	add_action( 'save_post', 'HM\\Redirects\\Admin_UI\\handle_redirect_saving', 13 );
+
+	return $post_id;
+}
+
+/**
+ * Clear all caches for memory management.
+ *
+ * From Automattic, name kept with love.
+ */
+function stop_the_insanity() {
```

Comment:
Please, please, please give this a proper function name. I do not want any of Automattic's beloved development anti-patterns anywhere. Their code bases are cluttered with non sense because a small subset of their developers think it's "cute".

Reply:
It does have a proper name. It's a homage to the original. :)

I will change it but for the record I think disparaging it because you see it as a "Automattic's beloved development anti-pattern" is lousy reasoning. You could have made your point without the dig at Automatic.